src/test/java/uk/gov/dhsc/htbhf/smartstub/controller/HMRCBenefitControllerIntegrationTest.java

```
@@ -0,0 +1,170 @@
+
+    private static final URI ENDPOINT = URI.create("/v1/hmrc/benefits");
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Test
+    void shouldReturnNoChildrenForMatchingNino() {
+        HMRCEligibilityRequest anEligibilityRequest = anHMRCEligibilityRequest();
+
+        ResponseEntity<BenefitDTO> benefit = restTemplate.postForEntity(ENDPOINT, anEligibilityRequest, BenefitDTO.class);
+
+        assertSuccessfulNumberOfChildrenResponse(benefit, 0, 0);
+    }
+
+    @Test
+    void shouldReturnTwoChildrenUnderOneForMatchingNino() {
+        PersonDTO person = aPersonWithChildrenUnderOne(2);
+        HMRCEligibilityRequest request = anHMRCEligibilityRequestWithPerson(person);
+
+        ResponseEntity<BenefitDTO> benefit = restTemplate.postForEntity(ENDPOINT, request, BenefitDTO.class);
+
+        assertSuccessfulNumberOfChildrenResponse(benefit, 2, 2);
+    }
+
+    @Test
+    void shouldReturnTwoChildrenUnderFourForMatchingNino() {
+        PersonDTO person = aPersonWithChildrenUnderFour(2);
+        HMRCEligibilityRequest request = anHMRCEligibilityRequestWithPerson(person);
+
+        ResponseEntity<BenefitDTO> benefit = restTemplate.postForEntity(ENDPOINT, request, BenefitDTO.class);
+
+        assertSuccessfulNumberOfChildrenResponse(benefit, 0, 2);
+    }
+
+    @Test
+    void shouldReturnNoMatchForMatchingNino() {
+        PersonDTO person = aPersonNotFound();
+        HMRCEligibilityRequest request = anHMRCEligibilityRequestWithPerson(person);
+
+        ResponseEntity<BenefitDTO> benefit = restTemplate.postForEntity(ENDPOINT, request, BenefitDTO.class);
+
+        assertSuccessfulStatusResponse(benefit, NOMATCH);
+        assertThat(benefit.getBody().getNumberOfChildrenUnderOne()).isNull();
+        assertThat(benefit.getBody().getNumberOfChildrenUnderFour()).isNull();
+    }
+
+    @Test
+    void shouldReturnSameChildrenUnder1AndChildrenUnder4WhenChildrenUnder1TooHigh() {
+        PersonDTO person = aPersonWithChildren(3, 1);
```

Do we not want to throw an exception this case?

---

I think we're better avoiding 'unexplained' errors in the front end - enter what seems to be a perfectly valid NINO and get a 500 error in response

---

This was discussed and done in a PR yesterday, the thought was that this check would restrict too many NINOs from being valid, so we'll be matching the under 4 and under 1 children instead and we now have a single NINO that causes an Exception to be thrown. Full details are in the README.md.